deploy/crds/logging.openshift.io_clusterlogforwarders_crd.yaml

```diff
@@ -0,0 +1,203 @@
+                      failures. Auto-reconnect is enabled by default.
+                    properties:
+                      firstDelayMilliseconds:
+                        description: FirstDelayMilliseconds is the time to wait after
+                          a disconnect before the first reconnect attempt. If reconnect
+                          fails, the delay is doubled on each subsequent attempt.
+                          The default is determined by the output type.
+                        format: int64
+                        type: integer
+                      maxDelayMilliseconds:
+                        description: MaxDelaySeconds is the maximum delay between
+                          failed re-connect attempts, and also the maximum time to
+                          wait for an unresponsive connection attempt. The default
+                          is determined by the output type.
+                        format: int64
+                        type: integer
+                      reliability:
+                        description: Reliability policy for data delivery after a
+                          re-connect.  This is simple short-hand for configuring the
+                          output to a given level of reliability.  The exact meaning
+                          depends on the output `type`.  The default is determined
+                          by the output type.
+                        enum:
+                        - Unreliable
+                        - Retry
+                        type: string
+                    type: object
+                  secret:
+                    description: "SecretRef refers to a `secret` object for secure
+                      communication. \n Client-authenticated TLS is enabled if the
+                      secret contains keys `tls.crt`, `tls.key` and `ca.crt`. Output
+                      types with password authentication will use keys `password`
+                      and `username`, not the exposed 'username@password' part of
+                      the `destination` URL."
+                    properties:
+                      name:
+                        description: Name is unique within a namespace to reference
+                          a secret resource.
+                        type: string
+                      namespace:
+                        description: Namespace defines the space within which the
+                          secret name must be unique.
+                        type: string
+                    type: object
+                  syslog:
+                    description: Syslog optional extra syslog-specific properties.
+                    properties:
+                      facility:
+                        description: Facility for outgoing syslog records. Special
+                          value 'COPY' means copy from incoming log record.
+                        enum:
```

Comment:
is the enum really necessary here?

Reply:
It gives us explicit openapi validation, which I think is better than hidden validation code or failing late when something tries to parse the string as a facility. Did you have some other approach in mind?

Reply:
I will ask the question given what I saw working with eclipse plugins and apis:  If we get out of sync with the possible values could that not lead to a validation failure because we are behind a version for a given syslog spec. This means we would unnecessarily reject a value.  Maybe not likely but just a thought.

Reply:
i'm not sure we even need to validate this, if a user specifies a value we should expect that they know which value they actually want

Reply:
I did hesitate over such a large enum - you've convinced me. 

Syslog severities and facilities are defined in the standard as integers, different syslog impls use a mostly-but-not-entirely-consistent set of spellings for names.  Instead of an advertised enum, we should probably accept integers and all capitalizations the common symbolic names.

IMO we should still check this during validation and normalize to whatever representation is used by the syslog library that we use internally, but I'll kill the openapi enum.